Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2453 @@
+ drop CryList.
+ CryList.0 = Cryptos
+ do j = 1 to Cryptos
+    nextEntryHandle = JSON_getArrayEntry(ArrayResponse,j-1)
+    CryList.j.num=JSON_findValue2(nextEntryHandle,"number")
+    CryList.j.type=JSON_findValue(nextEntryHandle,"activation-type",,
+                        HWTJ_STRING_TYPE)
+
+    say CryList.j.num || ' , ' || Crylist.j.type
+                          
+    if j=1 then
+      REXXWRT.outLC = ',,,'||CryList.j.num||','||CryList.j.type
+    else
+      REXXWRT.outLC = REXXWRT.outLC||','||CryList.j.num||','||CryList.j.type
+
+ end /* array entries */
+ end /* Cryptos  */
+
+ return 0 /* end function */
+
+/*******************************************************/
+/* Function: getAssignedDomains                        */
+/*                                                     */
+/* Retrieve content returned from the assigned crypto  */
+/* domains value, which is an array of zero or more    */
+/* objects.                                            */
+/* Store the specific content into REXXWRT stem.       */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+getAssignedDomains:
+ ArrayResponse = arg(1)
+
+ keyType = 'assigned-crypto-domains'
+
+ /* First assess if this property was even found in the response */
+ if foundJSONValue(ArrayResponse) = FALSE then
+   do
+     say '** Unable to locate '||keyType||' in the response **'
+     return 0
+   end
+
+ /****************************************************************/
+ /* Determine the number of items in the array                   */
+ /****************************************************************/
+ Domains = JSON_getArrayDim(ArrayResponse)
+ if Domains = -1 then
+    return fatalError( '** Unable to retrieve crypto domains **' )
+ else if Domains = 0 then
+    say '** Empty list of crypto domains returned **'
+
+ /****************************************************************/
+ /* Traverse the array to populate domains list                  */
+ /****************************************************************/
+ If Domains > 0 then do
+   say 'Processing information for '||Domains||' crypto domains'
+   say 'Index , Mode'
+   outLC = outLC + 1
+   REXXWRT.outLC = ',,,'||keyType||' (Domain Index & Access Mode Array)'
+   outLC = outLC + 1
+ drop DmnsList.
+ DmnsList.0 = Domains
+ do k = 1 to Domains
+    nextEntryHandle = JSON_getArrayEntry(ArrayResponse,k-1)
+    DmnsList.k.index=JSON_findValue2(nextEntryHandle,"domain-index")
+    DmnsList.k.mode=JSON_findValue(nextEntryHandle,"access-mode",,
+                        HWTJ_STRING_TYPE)
+
+    say DmnsList.k.index || ' , ' || Dmnslist.k.mode
+                          
+    if k=1 then
+      REXXWRT.outLC = ',,,'||DmnsList.k.index||','||DmnsList.k.mode
+    else
+      REXXWRT.outLC = REXXWRT.outLC||','||DmnsList.k.index||','||DmnsList.k.mode
+
+ end /* array entries */
+ end /* domains  */
```

Comment:
fix indentation please (it's valid code just a bit hard to read :) )

Reply:
Done